dotnet/src/SFTPService/SftpService.cs

```diff
@@ -0,0 +1,122 @@
+        /// </summary>
+        /// <param name="localFilePath">is the source path of the file .</param>
+        /// <param name="remoteFilePath">is the destination path of the file .</param>
+        /// <returns>void</returns>
+        public void UploadFile(string localFilePath, string remoteFilePath)
+        {           
+            using var client = new SftpClient(_config.Host, _config.Port == 0 ? 22 : _config.Port, _config.UserName, _config.Password);
+            try
+            {
+                client.Connect();
+                using var s = File.OpenRead(localFilePath);
+                client.UploadFile(s, remoteFilePath);
+                _logger.LogInformation($"Finished uploading file [{localFilePath}] to [{remoteFilePath}]");
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed in uploading file [{localFilePath}] to [{remoteFilePath}]");
+            }
+            finally
+            {
+                client.Disconnect();
+            }
+        }
+
+        /// <summary>
+        /// download the files from remote location to local path
+        /// </summary>
+        /// <param name="localFilePath">is the source path of the file .</param>
+        /// <param name="remoteFilePath">is the destination path of the file .</param>
+        /// <returns>void</returns>
+        public void DownloadFile(string remoteFilePath, string localFilePath)
+        {
+            using var client = new SftpClient(_config.Host, _config.Port == 0 ? 22 : _config.Port, _config.UserName, _config.Password);
+            try
+            {
+                client.Connect();
+                using var s = File.Create(localFilePath);
+                client.DownloadFile(remoteFilePath, s);
+                _logger.LogInformation($"Finished downloading file [{localFilePath}] from [{remoteFilePath}]");
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed in downloading file [{localFilePath}] from [{remoteFilePath}]");
+            }
+            finally
+            {
+                client.Disconnect();
+            }
+        }
+
+        /// <summary>
+        /// delete the remote files 
+        /// </summary>     
+        /// <param name="remoteFilePath">is the path of the file to be deleted .</param>
+        /// <returns>void</returns>
+        public void DeleteFile(string remoteFilePath)
+        {
+            using var client = new SftpClient(_config.Host, _config.Port == 0 ? 22 : _config.Port, _config.UserName, _config.Password);
+            try
+            {
+                client.Connect();
+                client.DeleteFile(remoteFilePath);
+                _logger.LogInformation($"File [{remoteFilePath}] deleted.");
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, $"Failed in deleting file [{remoteFilePath}]");
```

Comment:
@paras-lalchandani Should be `Failed to delete file`

Reply:
@motymichaely done